internal/features/features.go

```diff
@@ -30,12 +30,21 @@ const (
 	// the last revision is still the same at the target repository,
 	// and if that is so, skips the reconciliation.
 	OptimizedGitClones = "OptimizedGitClones"
+	// ForceGoGitImplementation ignores the value set for gitImplementation
+	// and ensures that go-git is used for all GitRepository objects.
+	//
+	// When enabled, libgit2 won't be initialized, nor will any git2go cgo
+	// code be called.
+	ForceGoGitImplementation = "ForceGoGitImplementation"
 )
 
 var features = map[string]bool{
 	// OptimizedGitClones
 	// opt-out from v0.25
 	OptimizedGitClones: true,
+	// ForceGoGitImplementation
+	// opt-in from v0.32
+	ForceGoGitImplementation: false,
```

Comment:
Up for debate whether we want this to be opt-in or opt-out. We get most value with the latter, whilst still providing an easy way for users to revert back to the previous behaviour.

Reply:
After talk with some maintainers, I will revert this to be opt-out.